cli/compose/loader/merge.go

```diff
@@ -0,0 +1,108 @@
+package loader
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/docker/cli/cli/compose/types"
+	"github.com/imdario/mergo"
+	"github.com/pkg/errors"
+)
+
+type specials struct {
+	m map[reflect.Type]func(dst, src reflect.Value) error
+}
+
+func (s *specials) Transformer(t reflect.Type) func(dst, src reflect.Value) error {
+	if fn, ok := s.m[t]; ok {
+		return fn
+	}
+	return nil
+}
+
+func merge(configs []*types.Config) (*types.Config, error) {
+	base := configs[0]
+	for _, override := range configs[1:] {
+		var err error
+		// merge services
+		base.Services, err = mergeServices(base.Services, override.Services)
+		if err != nil {
+			return base, errors.Wrap(err, "cannot merge services")
```

Comment:
Errors here might be different to debug when there are many files because there is no mention of which file failed to merge. Maybe `types.Config` needs a `Filename` field so that it can be included here?